Code-the-Dream-School/cc-prac-team4-front
# PP-83: added functionality to login sign up page

**Pull Request Description:**

This pull request introduces a new feature that enhances the user authentication and registration process in our application. It includes the implementation of a login and registration form with improved user experience and error handling.

**Key Changes:**

1. Improved a LoginSignUp component that handles the login and registration functionality.
2. Implemented tab switching between the registration and login forms for easy user navigation: when the user is successfully registered the tab is switching from registration and login.
3. Utilized the react-bootstrap library to display success and error messages using the Alert component.
4. Integrated API calls for login and registration using the axios library.
5. Implemented a reducer function and dispatched actions to update the state based on the API responses.
6. Utilized cookies to store and retrieve the user's authentication token.
7. Added a proxy middleware sets up to forward requests from the '/api' endpoint to a target server at 'http://localhost:8000/'.

**Steps for QA Testing:**

1. Install: 
 > 1. `npm install axios@1.4.0 npm `
 > 2. `install http-proxy-middleware`
 > 3. `npm install js-cookie react-router-dom npm `
 > 4. `install react-bootstrap@latest bootstrap@latest`

2. Run the app `npm start`
3. Open the application and navigate to the login and registration pages: [http://localhost:3000/login](http://localhost:3000/login)
4. Verify that both the login and registration forms are displayed correctly.
5. Test switching between the login and registration tabs and ensure that the form content updates accordingly.
6. Submit the login form with valid credentials and verify that the user is successfully logged in and ensure that the user is redirected to the user profile page (http://localhost:3000/account).
7. Submit the login form with invalid credentials and ensure that the appropriate error message is displayed.
8. `Check that the user with name Joe Biden can register on the site by entering only his first name.`
9. Fill out the registration form with valid data and submit it to create a new user account.
10. Verify that the success message is displayed and the user is redirected to the login tab.
11. Attempt to register with an existing email address and verify that the error message is displayed.
12. Test form validation by leaving required fields empty and verify that the corresponding error messages are shown.
